internal/application/application.go

```diff
@@ -789,31 +828,25 @@ func fetch(ctx context.Context, cluster *kubernetes.Cluster, app *models.App) er
 	// straighten the workload structure a bit further.
 
 	app.Workload, err = NewWorkload(cluster, app.Meta, instances).Get(ctx)
-	return err
-}
-
-// calculateStatus sets the Status field of the App object.  To decide what the status
-// value should be, it combines various pieces of information, i.e. status of possible
-// staging, presence of a workload, etc.
-// - If Status is ApplicationError, leave it as it (it was set by "Lookup")
-// - If there is an active staging job, app is: ApplicationStaging
-// - If there is no active staging job and no workload, app is: ApplicationCreated
-// - If there is no active staging job and a workload, app is: ApplicationRunning
-func calculateStatus(ctx context.Context, cluster *kubernetes.Cluster, app *models.App) error {
-	if app.Status == models.ApplicationError {
-		return nil
+	if err != nil {
+		err = errors.Wrap(err, "workload loading")
+		app.StatusMessage = err.Error()
+		app.Status = models.ApplicationError
+		return err
 	}
+
 	staging, err := CurrentlyStaging(ctx, cluster, app.Meta.Namespace, app.Meta.Name)
```

Comment:
This is called per application, right ?
Does it cause a kube API call ?
If yes, could the information be pulled out of the list-loaded data ?
And if not, maybe out of a another list-load ?

Reply:
Ah yes, it does. I guess so, I'll have a look at it. :+1:

Reply:
good luck